### ISSUE TITLE
Adding new instructions

### ISSUE THREAD

**Issue by osta1** [openhwgroup/core-v-sw#24]
Hello, I would like to know how it is possible to add new instructions to pulp-riscv-gnu-toolchain. I have checked on google but I am not able to find some suitable answers. Thank you for informing me about the procedure. 
Much thanks!

**Comment by jeremybennett** [comment]
@osta1 Before I reply in detail, how much do you know about modifying the GNU tool chain?

**Comment by osta1** [comment]
Till now, I did not modify the toolchain; I have installed the pulp-riscv-gnu-toolchain and I have set the environment in order to run some c codes and assembly codes on CV32. At the instant, I want to add a new instruction (fast mul instruction).  On google I found the following tutorial : https://nitish2112.github.io/post/adding-instruction-riscv/.
But, this tutorial is applied for riscv toolchain not for pulp-riscv-gnu-toolchain; there are some minor differences between these two toolchains. Is there any tutorial similar to this one but for pulp-riscv-gnu-toolchain?

**Comment by osta1** [comment]
Hello Jeremy,

Thank you for your reply. I have checked the files, I think the two most important files are : riscv-opc.c and riscv-opc.h. I have tried to modify the files by adding the new instruction following the procedure of the previous tutorial. But I would like to know, where is located the opcodes repository in the new toolchain (pulp-riscv-gnu-toolchain) . Since in the new toolchain, I am not able to find the file riscv-opcodes/opcodes; in order to assign the instruction as done in the first step of the tutorial. 

Thank you,

Mario.
